classes/class4/README.md

```diff
@@ -1 +1,484 @@
+
+Some key trends to note include the rise of Chrome as well as the fall of IE and Firefox.
+Also worth noting is the rapid growth of mobile browsing (pink).
+We won't address [mobile web design](http://www.smashingmagazine.com/guidelines-for-mobile-web-development/) in this class,
+but it is definitely something to consider before launching a new site.
+
+Ok, back to the console.
+On any webpage, right-click, and select `Inspect Element`. You'll see something like this pop up in the bottom of the window:
+![devconsole](images/devconsole.png)
+The dev console contains a ton of features (seriously, there are more features than I've even heard of).
+We'll start with the basic tabs that you use most often:
+* Elements: This is a collapsable and editable view of the HTML on the current page.
+You can double-click to edit fields or [DOM](http://en.wikipedia.org/wiki/Document_Object_Model) nodes.
+On the right side is a panel showing the CSS styling applied to the element you are highlighting. Again, double-click to edit.
+A useful shortcut here is the magnifying glass in the top-left corner of the console.
+Click that and then click in the browser somewhere to jump to the HTML of that element.
+* Network: While you're in this tab, refresh the browser.
+The tab will list all the requests that the web page makes, including the size, timing, and specific response.
+This can be useful for debugging your server's responses or other API's.
+* Sources: The holy grail of JavaScript debugging tools. Set breakpoints, step through code,
+all the fun stuff that an IDE would give you. We'll get back to this when we go over debugging in more depth.
+* Console: A full JavaScript console that operates in the current scope of the Javascript on the page (including inside breakpoints).
+Run commands, and read through your `console.logs` in this window.
+As a bonus, if you click the "console icon" ![console icon](images/consoleicon.png) in the upper-right corner,
+the console will pop up below whatever other tab you want to reference.
+
+## HTML Reprise
+We've seen some HTML before but let's go through a more formal overview of it.
+[HTML](http://en.wikipedia.org/wiki/HTML) is a markup language that uses tagged elements
+to describe the structure of a webpage.
+Elements are specified with start and end tags which enclose their content,
+including children elements (`<div><p>Hi again</p></div>`).
+Some elements cannot have content or children and do not have an end tag.
+Instead their start tag ends with a slash (`<br/>`)
+While HTML can include presentation specifications,
+it is highly recommended to use HTML for structure and [Cascading Style Sheets](http://en.wikipedia.org/wiki/Cascading_Style_Sheets) (CSS)
+for define the look and layout of the page. There are *many* [HTML tags](https://developer.mozilla.org/en-US/docs/Web/HTML/Element),
+but you will mostly use a small number:
+* [`<a>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a):
```

Comment:
Maybe make it explicitly clear that these are links to the "official" documentation for them.